fsmonitor.c

```diff
@@ -32,17 +33,28 @@ int read_fsmonitor_extension(struct index_state *istate, const void *data,
 	uint32_t ewah_size;
 	struct ewah_bitmap *fsmonitor_dirty;
 	int ret;
+	struct strbuf last_update = STRBUF_INIT;
 
 	if (sz < sizeof(uint32_t) + sizeof(uint64_t) + sizeof(uint32_t))
 		return error("corrupt fsmonitor extension (too short)");
 
 	hdr_version = get_be32(index);
 	index += sizeof(uint32_t);
-	if (hdr_version != INDEX_EXTENSION_VERSION)
+	if (hdr_version == INDEX_EXTENSION_VERSION1) {
```

Comment:
@dscho Question for you on backwards compatibility.  When there is a new version of a hook like here, what is the usual way of handling that with git? Here I am forcing the version of the hook to be 2, but should I instead be doing something like adding a hook version to the config that defaults to 1 so that version of the hook will continue to work when users install a new version of git?  With it currently anyone with version 1 of the hook will fail until they have a version 2.

Reply:
> When there is a new version of a hook like here, what is the usual way of handling that with git?

I don't think that we have a precedent for that yet ;-)

> Here I am forcing the version of the hook to be 2, but should I instead be doing something like adding a hook version to the config that defaults to 1 so that version of the hook will continue to work when users install a new version of git?

I think we'll want to have a config variable, yes, if only to keep regression tests that make sure that Git can still talk to v1 fsmonitor hooks.

> With it currently anyone with version 1 of the hook will fail until they have a version 2.

How about doing a fall-back? Like, defaulting to `-1` for the version, overridable via the config, and if it is `-1`, try 2 first, and if that fails, try 1, and if that succeeds, set the global variable to 1, otherwise to 0 (and consistently skip the fsmonitor for the rest of the process' life time).

What do you think?

Reply:
>How about doing a fall-back?

That would be the most robust.  Thanks for the suggestion.